ethereum/blockstack/block_stack.go

```diff
@@ -1,13 +1,15 @@
-package blockwatch
+package blockstack
 
 import (
 	"sync"
 
+	"github.com/0xProject/0x-mesh/ethereum/miniheader"
 	"github.com/0xProject/0x-mesh/meshdb"
 )
 
-// Stack allows performing basic stack operations on a stack of meshdb.MiniHeaders.
-type Stack struct {
+// BlockStack allows performing basic stack operations on a stack of meshdb.MiniHeaders stored
+// in the DB backing our meshdb package.
+type BlockStack struct {
```

Comment:
Optionally, maybe we could call this `DBStack` to more clearly indicate how this implementation is different from `SimpleStack`?